### PR TITLE
Bluetooth: controller: Fixes for CIS Central error handling

### PR DESCRIPTION
**Pull request by mtpr-ot**
- Fix masking for no coded phy
- Allow ISO_Interval < SDU_Interval for framed mode
- Change BT_HCI_ERR_INSUFFICIENT_RESOURCES to BT_HCI_ERR_CONN_LIMIT_EXCEEDED.
- Prevent starting same CIS twice
- Cancel an initiated CIS creation procedure if terminated before sending CIS_IND.
- Implement canceling of local CC procedure. Respond to CIS_RSP with REJECT, if canceled after CIS_REQ was sent.
- Introduce state CIG_STATE_INITIATING for central, to keep track of initiating CIS connection, in transition between CONFIGURABLE and ACTIVE.

Fixes EBQ test /HCI/CIS/BC-03-C.

### CODE REVIEW

**Review by erbr-ot — APPROVED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c [hunk 1]
```diff
@@ -267,7 +273,8 @@ uint8_t ll_cig_parameters_commit(uint8_t cig_id, uint16_t *handles)
 			if (!cis) {
 				/* No space for new CIS */
 				ll_iso_setup.cis_idx = 0U;
-				err = BT_HCI_ERR_INSUFFICIENT_RESOURCES;
+
+				err = BT_HCI_ERR_CONN_LIMIT_EXCEEDED;
 				goto ll_cig_parameters_commit_cleanup;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_central_iso.c:278]:
```suggestion
				err = BT_HCI_ERR_CONN_LIMIT_EXCEEDED;
				
				goto ll_cig_parameters_commit_cleanup;
```

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c [hunk 2]
```diff
@@ -642,7 +649,13 @@ uint8_t ll_cis_create_check(uint16_t cis_handle, uint16_t acl_handle)
 
 		/* Verify handle validity and association */
 		cis = ll_conn_iso_stream_get(cis_handle);
-		if (cis->lll.handle == cis_handle) {
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_central_iso.c:652]:
```suggestion
```

**Review by kruithofa — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]
Minor nitpicks, otherwise LGTM.

**Pull request merged**